patroni/utils.py

```diff
@@ -586,7 +590,7 @@ def polling_loop(timeout: Union[int, float], interval: Optional[Union[int, float
     while time.time() < end_time:
         yield iteration
         iteration += 1
-        time.sleep(interval)
+        time.sleep(float(interval))
 
 
 def split_host_port(value: str, default_port: int) -> Tuple[str, int]:
```

Comment:
```suggestion
def split_host_port(value: str, default_port: Optional[int]) -> Tuple[str, int]:
```

Some methods, like `__initialize` from `patroni.api.RestApiServer`, call `split_host_port` with `port=None`.